src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java

```diff
@@ -53,9 +53,12 @@ protected ReplayFile call() throws Exception {
                 preStatus = provider.getPreStatus(replay);
             } catch (MpqException e) {
                 LOG.warn("Could not parse replay, deferring to upload." + take, e);
-                preStatus = Status.NEW;
+
+                preStatus = Status.EXCEPTION;
             }
-            if (preStatus == Status.UPLOADED || preStatus == Status.UNSUPPORTED_GAME_MODE) {
+            if (preStatus == Status.UPLOADED ||
+                preStatus == Status.UNSUPPORTED_GAME_MODE ||
+                preStatus == Status.EXCEPTION) {
```

Comment:
I initially put that in, because the upload of a non-existent file caused some exception on the upload. But a simple file existence check in the service will do as well.